### ISSUE TITLE
Create a GitHub action to build PortAudio using the MSVC .vcproj file in build/msvc

### ISSUE THREAD

**Issue by RossBencina** [PortAudio/portaudio#368]
In order to avoid issues with people checking in broken code, we need an action that builds PortAudio using the .vcproj file.

I have created such a build action:

https://github.com/RossBencina/portaudio/blob/MSBuild-action/.github/workflows/MSBuild.yml

you can see it running here:

https://github.com/RossBencina/portaudio/actions/runs/435113591

---I am waiting for the build to be clean before I merge it. Depends on #363 and #364--- (both now merged, good to go).

**Comment by MichalPetryka** [comment]
You probably should also specify /p:Platform= for MSBuild and build both Win32 and x64

**Comment by RossBencina** [comment]
@MichalPetryka will that work if the project doesn't have an x64 target?

**Comment by RossBencina** [comment]
Note to self: Slow time to run `devenv.exe` (3.5 minutes) is possibly due to this issue: https://github.com/actions/virtual-environments/issues/128

May be better to use VCUpgrade.exe located in $(VSInstallDir)\common7\Tools see  https://devblogs.microsoft.com/cppblog/visual-studio-2010-c-project-upgrade-guide/

**Comment by MichalPetryka** [comment]
PortAudio does target x64 though
```
		<Platform
			Name="Win32"
		/>
		<Platform
			Name="x64"
		/>
```

**Comment by RossBencina** [comment]
> You probably should also specify /p:Platform= for MSBuild and build both Win32 and x64

Done.

**Comment by MichalPetryka** [comment]
Another thing worth testing would be building with both Unicode and Multi-Byte Character Set, though i have no idea how to control that from the commandline.

**Comment by RossBencina** [comment]
>  Unicode and Multi-Byte Character Set

Maybe but I don't think so. If you follow this logic to its conclusion we should also be building and running all the tests.

The purpose of this build action is simply to test that the visual studio project file works in all of its standard configurations.  For the purpose of testing the project file, we make no guarantees that arbitrary changes can be made to the project file. Altering unicode settings is not something we expect to be changed. The user shouldn't be manipulating the unicode settings when using the visual studio project -- so that's beyond the scope of testing the project file. 

Of course it has been established that the PortAudio code should be immune to Unicode settings (it should return UTF-8 strings irrespective of Windows configuration unicode #defines). It would be nice to test this somewhere, but I don't think it's relevant to testing the project file.

That's my take on it anyway.

**Comment by jmelas** [comment]
Great initiative! Two comments:
UNICODE should be the default for new development ie define _UNICODE and UNICODE
We have to define what the minimum version of Windows we support ie define _WIN32_WINNT

**Comment by MichalPetryka** [comment]
Yeah, I agree that UNICODE should be recommended, there are some issues with that though:
1. UNICODE is NT+, does PortAudio still support 95 and 98 with MME?
2. ASIO SDK doesn't compile under UNICODE, you have to #undef it at the start of one of files
3. MINGW seems to default to MBCS since sizeof(TCHAR) is 1

**Comment by RossBencina** [comment]
If you think that UNICODE is not correctly defined in the msvc project file please open a new issue. This issue is for the GitHub action only.

**Comment by jmelas** [comment]
Yes, I was refering to the GiHub action. For the msvc project I think it should better be replaced with CMake.
About explicit UNICODE: ok but we should remove all instances of TCHAR from our code. Luckily it seems only wmme uses TCHAR!
